### PR TITLE
First round of refactors to use new AtlasShellToolsCommandTemplate

### PR DESCRIPTION
**Pull request by lucaspcram**
### Description:
Some refactors to make use of the new template feature. This PR is not the complete refactor. A followup will address the now-deprecated template classes and fully convert the rest of the commands. For now, this PR should serve as a model for any downstream commands which would like to refactor to clean up their class hierarchies.

### Potential Impact:
Should be none, other than some template classes now deprecated.

### Unit Test Approach:
Added a new template test.

### Test Results:
Test suite 👍

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/SubAtlasCommand.java [hunk 1]
```diff
@@ -137,26 +120,26 @@ protected void processAtlas(final Atlas atlas, final String atlasFileName,
 
         if (this.optionAndArgumentDelegate.hasOption(SLICE_FIRST_OPTION_LONG))
         {
-            if (this.polygon.isPresent())
+            if (this.polygon != null)
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/SubAtlasCommand.java:123]: That seems odd, is it expected?

File: src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/templates/MultipleOutputCommand.java [hunk 2]
```diff
@@ -16,7 +16,9 @@
  * registers a '--output' option, the argument to which will be used as an output directory.
  *
  * @author lcram
+ * @deprecated
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/templates/MultipleOutputCommand.java:19]: Maybe say here what to use instead?

File: config/checkstyle/suppressions.xml [hunk 3]
```diff
@@ -13,6 +13,7 @@
     <suppress checks="AvoidEscapedUnicodeCharacters" files="src/main/java/org/openstreetmap/atlas/utilities/command/terminal/TTYAttribute.java" />
     <suppress checks="ParameterName" files="src/main/java/org/openstreetmap/atlas/utilities/function/TernaryFunction.java" />
     <suppress checks="ParameterName" files="src/main/java/org/openstreetmap/atlas/utilities/function/TernaryOperator.java" />
+    <suppress checks="ParameterName" files="src/main/java/org/openstreetmap/atlas/utilities/function/TernaryConsumer.java" />
```
Comment (matthieun) [inline config/checkstyle/suppressions.xml:16]: As I was commenting about checkstyle and the parameter names, this appeared hahaha

Comment (matthieun) [reply config/checkstyle/suppressions.xml:16]: Maybe we should just call them `one` `two` and `three` and not have to do that every time?

File: src/main/java/org/openstreetmap/atlas/utilities/function/TernaryConsumer.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+     *             if {@code after} is null
+     */
+    default TernaryConsumer<T, U, V> andThen(
+            final TernaryConsumer<? super T, ? super U, ? super V> after)
+    {
+        Objects.requireNonNull(after);
+        return (l, c, r) ->
+        {
+            accept(l, c, r);
+            after.accept(l, c, r);
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/utilities/function/TernaryConsumer.java:55]: How far until there is a stack overflow? 😺

**Review by matthieun — CHANGES_REQUESTED** [review]
Nice work! Just a few minor comments and questions!

File: src/main/java/org/openstreetmap/atlas/utilities/function/TernaryConsumer.java (hunk 4 above)
Comment (lucaspcram) [reply src/main/java/org/openstreetmap/atlas/utilities/function/TernaryConsumer.java:55]: Far enough that the Java standard library doesn't mind implementing it this way either :P

You can look at their `BiConsumer` for an example.

**Comment by lucaspcram** [comment]
Nice comments thanks @matthieun I will take a look and address them soon!

File: src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/SubAtlasCommand.java (hunk 1 above)
Comment (lucaspcram) [reply src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/SubAtlasCommand.java:123]: It is. Before, I was storing this instance variable as an `Optional` which is bad style. I changed it to a plain polygon, so now it needs a null check.

File: config/checkstyle/suppressions.xml (hunk 3 above)
Comment (matthieun) [reply config/checkstyle/suppressions.xml:16]: Wouls still be nice to get rid of the suppressions, though..

**Review by matthieun — APPROVED** [review]

**Pull request merged**